packages/nimble-components/src/utilities/unit-format/unit-scale/base/unit-scale.ts

```diff
@@ -0,0 +1,67 @@
+import type { ScaledUnit } from './scaled-unit';
+
+interface ScaledNumber {
+    readonly scaledValue: number;
+    readonly scaledUnit: ScaledUnit;
+}
+
+/**
+ * A unit scale consisting of a set of scaled units.
+ */
+export abstract class UnitScale {
+    public readonly baseScaledUnit: ScaledUnit;
+
+    public constructor(
+        public readonly supportedScaledUnits: readonly ScaledUnit[]
+    ) {
+        const unitsSorted = supportedScaledUnits.every(
+            (curr, i, arr) => i === 0 || arr[i - 1]!.scaleFactor < curr.scaleFactor
+        );
+        if (!unitsSorted) {
+            throw new Error(
+                'Supported scaled units must have unique and ordered scale factors'
+            );
+        }
+        const baseScaledUnit = supportedScaledUnits.find(x => x.isBase());
+        if (!baseScaledUnit) {
+            throw new Error(
+                'Supported scaled units must include a base scaled unit (scale factor=1)'
+            );
+        }
+        this.supportedScaledUnits = supportedScaledUnits;
+        this.baseScaledUnit = baseScaledUnit;
+    }
+
+    // Note that for the sake of reducing complexity in the implementation,
+    // we do NOT consider the effects of rounding when picking the unit to
+    // use for a given value. If formatting results in rounding, a value
+    // may be shown with an unexpected unit. Examples:
+    // - 999 bytes with two significant digits => "1000 bytes" (instead of "1 kB")
+    // - 0.00000000000000001 volts (= 0.01 fV) with one fractional digit => "0 fV" (instead of "0 volts")
```

Comment:
Could you make sure we have test cases that verify the examples provided here (or something equivalent)?

Reply:
Since this is a behavior that we are settling for, rather than one that we explicitly want, I didn't think it made sense to assert it via tests. But I defer to your judgment.

Reply:
Makes sense to test our edge cases so the behavior is known and it is clear if it changes rather than untested and unknown